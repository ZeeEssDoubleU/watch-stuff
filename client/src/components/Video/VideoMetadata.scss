@import "../../styles/shared.scss";

.video-metadata {
	color: #707070;
	font-size: 16px;

	h3 {
		font-weight: 400;
		color: $text-color-dark;
	}

	.video-stats {
		display: flex;
		justify-content: space-between;
		width: 100%;
		align-items: center;

		.video-actions {
			display: flex;
			align-items: center;

			.button {
				height: 30px;

				span {
					margin-left: 5px;
				}

				.highlight {
					color: red;
				}

				.icon {
					transition: transform 0.2s;
				}

				&:hover {
					.icon {
						transform: scale(1.2);
					}
				}
			}

			.ellipsis {
				padding: 0 5px;
				border: none;
				box-shadow: none;
			}

			& > * {
				margin-left: 20px;
				margin-right: 0;
			}
		}
	}
}
