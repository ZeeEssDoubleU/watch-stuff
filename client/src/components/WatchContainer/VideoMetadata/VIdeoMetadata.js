import React from "react";
import PropTypes from "prop-types";
import { Button, Divider, Icon } from "semantic-ui-react";

import "./VideoMetadata.scss";
import Ratings from "../../Ratings/Ratings";

const VideoMetadata = props => {
	const viewCount = Number(props.viewCount).toLocaleString() || "";

	return (
		<div className="video-metadata">
			<h3>Video Title</h3>
			<div className="video-stats">
				<span>{viewCount} views</span>
				<div className="video-actions">
					<Ratings likes={75} dislikes={25} />
					<Button basic icon labelPosition="left">
						<Icon name="share" />
						SHARE
					</Button>
					<Button basic icon labelPosition="left">
						<Icon name="add circle" />
						SAVE
					</Button>
					<Button basic icon>
						<Icon name="ellipsis horizontal" />
					</Button>
				</div>
			</div>
			<Divider />
		</div>
	);
};

VideoMetadata.propTypes = {};

export default VideoMetadata;
