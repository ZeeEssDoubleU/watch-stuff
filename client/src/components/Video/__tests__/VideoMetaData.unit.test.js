import React from "react";
import { shallow } from "enzyme";

import VideoMetadata from "../VideoMetadata";
import watchDetails from "./props/watchDetails";

describe("VideoMetadata", () => {
	test("renders without props", () => {
		const wrapper = shallow(<VideoMetadata />);
		expect(wrapper).toMatchSnapshot();
	});
	test("renders with video prop", () => {
		const wrapper = shallow(<VideoMetadata video={watchDetails} />);
		expect(wrapper).toMatchSnapshot();
	});
});
