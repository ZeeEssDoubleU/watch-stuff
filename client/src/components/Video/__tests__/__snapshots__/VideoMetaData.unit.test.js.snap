// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata renders with video prop 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Overwatch Animated Short | “Reunion”
  </h3>
  <div
    className="video-stats"
  >
    <span>
      2,221,089
       views
    </span>
    <div
      className="video-actions"
    >
      <Ratings
        dislikes={1524}
        likes={141186}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        SHARE
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="add circle"
        />
        SAVE
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata renders without props 1`] = `<div />`;
